### DISCUSSION TITLE
How to detect black on black text ? Please help

### DISCUSSION THREAD

**Discussion by raghavzns** [PaddlePaddle/PaddleOCR#13724] [Q&A]
Hi All,
I have to detect black text on black background , can anyone please help how to perform this using paddleOCR. PaddleOCR isnt working at all on black on black. Below is the reference  image
![t1](https://github.com/user-attachments/assets/176d56c8-d799-4697-8bfe-6ca2362f9d8f)

**Comment by GreatV** [comment]
1. Preprocessing the Image
Contrast Enhancement: Increase the contrast between the text and background using techniques like histogram equalization, adaptive histogram equalization (CLAHE), or other contrast enhancement methods.
Edge Detection: Apply edge detection algorithms like Canny or Sobel to highlight the text edges, which may help in differentiating the text from the background.
Inversion: Invert the colors of the image so that the black text becomes white on a white background. This might make the text more detectable.
2. Custom Model Training
If preprocessing does not yield satisfactory results, consider fine-tuning a pre-trained PaddleOCR model on a dataset specifically created for black-on-black text. This dataset would consist of similar images with black text on black backgrounds, annotated accordingly.
